src/main/java/core/mate/academy/service/BulldozerProducer.java

```diff
@@ -0,0 +1,21 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BulldozerProducer implements MachineProducer<Machine> {
```

Comment:
The `BulldozerProducer` should be parameterized with `Bulldozer` instead of `Machine` to prevent the creation of a producer that does not use `Machine` or its subtypes.